go/types/sequence_cursor.go

```diff
@@ -19,10 +19,13 @@ func newSequenceCursor(parent *sequenceCursor, seq sequence, idx int) *sequenceC
 		d.PanicIfFalse(idx >= 0)
 	}
 
-	return &sequenceCursor{parent, seq, idx}
+	cur := &sequenceCursor{parent, seq, idx}
+	d.Chk.NotNil(cur.seq)
```

Comment:
did you do this or did I? also, in either case, why no check for nil above at the beginning?